js/reducers/filters.js

```diff
@@ -1,15 +1,24 @@
-import { MESSAGES_FILTER } from '../constants/AppConstants';
+import { MESSAGES_FILTER, MESSAGES_FILTER_CLEAR } from '../constants/AppConstants';
 
 // @TODO: Set initial state based on Services.prefs value for severity filters
 //        Set current pref based on changed severity filters
 
 function filters(state = {}, action) {
   Object.freeze(state); // Don't mutate state directly, always use assign()!
+  let newState = Object.assign({}, state);
 
   switch (action.type) {
+    case MESSAGES_FILTER_CLEAR:
```

Comment:
If I'm thinking of things correctly, we don't need to Object assign and set props manually for this case. We should be able to just return an empty object. I might be wrong about that, though.